### PR TITLE
Eslint

### PR DESCRIPTION
**Pull request by Droogans**
JIRA: https://jira.rax.io/browse/FRMW-727, https://jira.rax.io/browse/FRMW-728, https://jira.rax.io/browse/FRMW-730, https://jira.rax.io/browse/FRMW-731, https://jira.rax.io/browse/FRMW-733

### LGTMs
- [x] Dev LGTM
- [x] Dev/QE LGTM

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -27,39 +28,40 @@
...
+        "max-len": [2, 120, { "ignoreUrls": true }],
+        "new-cap": 0,
+        "no-bitwise": 2,
+        "no-caller": 2,
+        "no-empty": 2,
         "no-multiple-empty-lines": [2, { "max": 2 }],
+        "no-undef": 2,
+        "no-unused-expressions": [2, { "allowTernary": true, "allowShortCircuit": true }],
+        "no-unused-vars": 2,
+        "no-use-before-define": [2, { "functions": false }],
```
Comment (CITguy) [inline .eslintrc.json]: This is to allow function hoisting, correct?

File: package.json [hunk 2]
```diff
@@ -14,8 +14,8 @@
   "scripts": {
     "test": "grunt test",
     "postinstall": "grunt hooks",
-    "eslint-dev": "eslint -c .eslintrc.json --ignore-path .eslintignore src/**",
-    "eslint-test": "eslint -c .eslintrc.spec.json --ignore-path .eslintignore-spec src/**"
+    "eslint-dev": "eslint -c .eslintrc.json --ignore-path .eslintignore src/** demo/**",
+    "eslint-test": "eslint -c .eslintrc.spec.json --ignore-path .eslintignore-spec src/** demo/**"
```
Comment (CITguy) [inline package.json]: You can probably simplify the `.eslintignore` if you focus on `demo/scripts/**`. There shouldn't be scripts anywhere else in `demo/`.

File: .eslintrc.json (hunk 1 above)
Comment (Droogans) [inline .eslintrc.json]: Yep. I had to switch a few definitions to use hoisting.

**Comment by 100stacks** [comment]
Other than @CITguy's comments...nice initial steps with ESLint.

 LGTM :+1:

File: package.json (hunk 2 above)
Comment (CITguy) [inline package.json]: This is non blocking, btw. We can revisit the specifics around the blob when we integrate into grunt.

**Comment by CITguy** [comment]
LGTM

**Pull request merged**